dashboard/app/controllers/reports_controller.rb

```diff
@@ -75,17 +75,19 @@ def search_for_teachers
     end
 
     # Prune the set of fields to those that will be displayed.
+    @teacher_limit = 100
     @headers = ['ID', 'Name', 'Email', 'Address', 'Num Students']
-    @teachers = @teachers.pluck('id', 'name', 'email', 'full_address', 'COUNT(followers.id) AS num_students')
+    @teachers = @teachers.limit(@teacher_limit).pluck('id', 'name', 'email', 'full_address', 'COUNT(followers.id) AS num_students')
```

Comment:
Given that I'd like to display 100 results (if they exist), I need more than that in the initial query since the count may be chopped down by PD / no-PD restrictions.

Here my assumption is that there is some non-negligible cost to not limiting the initial query. If that is not the case, I'll simply remove the initial limit.